File: Program's_Contributed_By_Contributors/Java_Programs/Data_structure/Stack reverse.java
import java.util.Stack;
 
class Test {
 
    // using Stack class for
    // stack implementation
    static Stack<Character> st = new Stack<>();
 
    // Below is a recursive function
    // that inserts an element
    // at the bottom of a stack.
    static void insert_at_bottom(char x)
    {
 
        if (st.isEmpty())
            st.push(x);
 
        else {
 
            // All items are held in Function
            // Call Stack until we reach end
            // of the stack. When the stack becomes
            // empty, the st.size() becomes 0, the
            // above if part is executed and
            // the item is inserted at the bottom
            char a = st.peek();
            st.pop();
            insert_at_bottom(x);
 
            // push allthe items held
            // in Function Call Stack
            // once the item is inserted
            // at the bottom
            st.push(a);
        }
    }
 
    // Below is the function that
    // reverses the given stack using
    // insert_at_bottom()
    static void reverse()
    {
        if (st.size() > 0) {
 
            // Hold all items in Function
            // Call Stack until we
            // reach end of the stack
            char x = st.peek();
            st.pop();
            reverse();
 
            // Insert all the items held
            // in Function Call Stack
            // one by one from the bottom
            // to top. Every item is
            // inserted at the bottom
            insert_at_bottom(x);
        }
    }
 
    // Driver Code
    public static void main(String[] args)
    {
 
        // push elements into
        // the stack
        st.push('1');
        st.push('2');
        st.push('3');
        st.push('4');
 
        System.out.println("Original Stack");
 
        System.out.println(st);
 
        // function to reverse
        // the stack
        reverse();
 
        System.out.println("Reversed Stack");
 
        System.out.println(st);
    }
}

File: Program's_Contributed_By_Contributors/All_About_DSA/DataStructuresAndAlgortihms/Stack_Reverse.c
#include <stdio.h>
int push(int a[],int size,int element){
    a[size] = element;
    return 0;
}

int pop(int a[],int size,int n){
    return a[n-size];
}
int main(){
    int s;
    printf("Enter a number : ");
    scanf("%d",&s);
    int temp,n=0;
    temp = s;
    while (temp>0){
        n++;
        temp = temp/10;
    }
    temp = s;
    int stack[100];
    int element;
    for (int i=0;i<n;i++){
        element = temp%10;
        push(stack,i,element);
        temp = temp/10;
    }
    int a[n];
    int b = 0;
    for (int i=0;i<n;i++){
        push(a,i,pop(stack,i,n-1));
    }
    for (int i=0;i<n;i++){
        b = 10*b+pop(a,i,n-1);
    }
    printf("Original Number : %d\n",s);
    printf("Reversed Number : %d\n",b);
    return 0;
}